app/javascript/components/community/video-grid/useVideoGrid.tsx

```diff
@@ -32,37 +60,61 @@ export function useVideoGrid(
     setQuery,
   } = useList(videoRequest)
 
-  const { status, resolvedData, latestData, isFetching } =
+  const { resolvedData, latestData, isFetching } =
     usePaginatedRequestQuery<APIResponse>(
-      ['community-video-grid-key', request],
+      [
+        'community-video-grid-key',
+        request.query.criteria,
+        request.query.videoTrackSlug,
+        request.query.videoPage,
+      ],
       request
     )
 
+  // don't refetch everything with an empty criteria after mounting
+  const didMount = useRef(false)
   useEffect(() => {
+    if (!didMount.current) {
+      didMount.current = true
+      return
+    }
+
     const handler = setTimeout(() => {
       if (criteria.length > 2 || criteria === '') {
         setRequestCriteria(criteria)
+        setQuery({ ...request.query, criteria })
       }
     }, 500)
 
     return () => {
       clearTimeout(handler)
     }
-  }, [setRequestCriteria, criteria])
-
-  useHistory({ pushOn: removeEmpty(request.query) })
+  }, [criteria])
 
   const handleTrackChange = useCallback(
-    (track) => {
+    (track: VideoTrack) => {
       setPage(1)
       setCriteria('')
       setSelectedTrack(track)
 
-      setQuery({ ...request.query, trackSlug: track.slug, page: 1 })
+      setQuery({
+        ...request.query,
+        videoTrackSlug: track.slug,
+        videoPage: 1,
+      })
     },
     [setPage, setQuery, request.query]
   )
 
+  useQueryParams(request.query)
+
+  const handlePageTurn = useCallback(
```

Comment:
Any reason not to use:

```suggestion
  const handlePageChange = useCallback(
```

which seems more in line with what we're doing elsewhere